### DISCUSSION TITLE
Use of regular expressions instead of cucumber expressions

### DISCUSSION THREAD

**Discussion by phillidgithub** [Tallyb/cucumber-playwright#823] [General]
Hi, does anyone know how to use regular expressions in place of cucumber expressions in step definitions? I tried using ^some regex$ but have not had any success.

The cucumber docs suggest this is possible but I can't seem to make it work when using the cucumber-playwright setup defined in this repos.

https://github.com/cucumber/cucumber-expressions#readme
On platforms that don't have a literal syntax for regular expressions (such as Java), Cucumber will create a Cucumber Expression by default. To use Regular Expressions, add anchors (starting with ^ and ending with $) or forward slashes (/).

**Comment by Tallyb** [comment]
PW-Cucumber is using cucumber for that. nothing different. 
```
When(/^User creates new (foo|bar|baz) song named "([^"]+)" from scratch$/, async function (this: ICustomWorld, type: string, name: string) {
```

**Reply by phillidgithub** [reply]
Thank you very much. I didn't realise that forward slashes rather than quotes were required before and after the regex start and end anchors
